frontend/src/views/home/components/games/useFetchGames.ts

```diff
@@ -0,0 +1,17 @@
+import { fetchGames } from '@/infrastructure/games/fetchGames'
+import { useQuery } from '@tanstack/react-query'
+
+export default function useFetchGames() {
+  const { isLoading, error, data, isFetching } = useQuery({
+    queryKey: ['games'],
+    queryFn: () => fetchGames(),
+  })
+
+  return {
+    isLoading,
+    error,
+    games: data ?? [],
+    hasGames: (data?.length as number) > 0,
```

Comment:
I would avoid assertion here. Let us guarantee this will never be null or undefined:

```suggestion
    hasGames: (data ?? []).length > 0,
```